unicode-data/lib/Unicode/Internal/Char/UnicodeData/Decompositions.hs

```diff
@@ -2080,3 +2078,4 @@ decompose = \case
   '\195099' -> "\40726"
   '\195100' -> "\40763"
   '\195101' -> "\173568"
+  c -> [c]
```

Comment:
Though I am not against this change, the idea was that we should never call this if the character is not decomposable, and if we do this will fail and we will get to know, just to ward off any perf impact due to calling it unnecessarily.

The benefit of making it total is that we can call it unconditionally if we want to in some case.

Reply:
We should avoid partial functions, especially if has (undocumented) impact for the public API.